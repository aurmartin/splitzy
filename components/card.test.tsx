import Card from "@/components/card";
import { fireEvent, render } from "@testing-library/react-native";
import * as React from "react";
import { Text } from "react-native";

it("renders correctly with children", () => {
  const tree = render(
    <Card>
      <Text>Card content</Text>
    </Card>,
  );
  expect(tree).toMatchSnapshot();
});

it("handles onPress events", () => {
  const onPressMock = jest.fn();
  const { getByText } = render(
    <Card onPress={onPressMock}>
      <Text>Clickable card</Text>
    </Card>,
  );

  fireEvent.press(getByText("Clickable card"));
  expect(onPressMock).toHaveBeenCalled();
});

it("renders as View when no onPress provided", () => {
  const tree = render(
    <Card>
      <Text>Static card</Text>
    </Card>,
  );
  expect(tree).toMatchSnapshot();
});

it("renders as Pressable when onPress provided", () => {
  const tree = render(
    <Card onPress={() => {}}>
      <Text>Clickable card</Text>
    </Card>,
  );
  expect(tree).toMatchSnapshot();
});
