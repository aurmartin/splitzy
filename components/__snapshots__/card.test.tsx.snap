// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as Pressable when onPress provided 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      [
        {
          "backgroundColor": "white",
          "borderRadius": 8,
          "overflow": "hidden",
          "padding": 14,
        },
        undefined,
      ],
      {
        "overflow": "hidden",
      },
    ]
  }
>
  <Text>
    Clickable card
  </Text>
</View>
`;

exports[`renders as View when no onPress provided 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 8,
        "padding": 14,
      },
      undefined,
    ]
  }
>
  <Text>
    Static card
  </Text>
</View>
`;

exports[`renders correctly with children 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 8,
        "padding": 14,
      },
      undefined,
    ]
  }
>
  <Text>
    Card content
  </Text>
</View>
`;
